src/main/java/edu/ucsb/nceas/mdqengine/Worker.java

```diff
@@ -127,12 +132,69 @@ public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProp
                     return;
                 }
 
-                Worker wkr = new Worker();
-                String runXML = null;
                 String metadataPid = qEntry.getMetadataPid();
                 String suiteId = qEntry.getQualitySuiteId();
                 SystemMetadata sysmeta = qEntry.getSystemMetadata();
+                String nodeId = qEntry.getMemberNode();
+
+                // see if there is an existing run
+                // TODO: this is where some try counter logic comes in maybe
+                Run run = null;
+                try {
+                    run = Run.getRun(metadataPid, suiteId);
+                } catch (MetadigException me) {
+                    log.info("No existing found for pid: " + metadataPid + "suite: " + suiteId);
+                } catch (ConfigurationException ce) {
+                    log.error("Configuration exception");
+                }
+
+                if (run == null) {
+                    run = new Run();
+                    // if no run exists
+                    run.setObjectIdentifier(metadataPid);
+                    run.setSuiteId(suiteId);
+                    run.setNodeId(nodeId);
+                    run.setErrorDescription("");
+                    run.setRunCount(0);
+                }
+
+                // set the run_count to count + 1
+                Integer runCount = run.getRunCount() + 1;
+                run.setRunCount(runCount);
+
+                // log and exit if 10 attempts have been made
+                if (run.getRunCount() > 10) {
```

Comment:
I think this is better at the beginning of the attempt than at the end, since we aren't sure what would be causing runs to fail that many times in the first place. if it continually failed before getting to the final `if (runCount)...` logic, it would always keep trying. marking it failed 24 hours after the last attempt seems better to me than potentially never marking it as failed at all